### ISSUE TITLE
Trouble publishing e2e report when e2e job fails

### ISSUE THREAD

**Issue by lanej0** [woothemes/woocommerce#48296]
A number of the e2e reports aren't generating properly over in [WooCommerce Test Reports](https://woocommerce.github.io/woocommerce-test-reports/).  I've done some investigation, and it appears that the problem is originating in the monorepo.

For failed e2e test runs:

![Screenshot 2024-06-07 at 3 52 44 PM](https://github.com/woocommerce/woocommerce/assets/80276/abcc8a0c-47c4-4a65-8e63-4c812d8bd5af)

The step "Publish report to dashboard" fails.  It looks like it isn't getting the GH_TOKEN set properly (it looks like it's blank in that screenshot).  This comes from [this run](https://github.com/woocommerce/woocommerce-test-reports/actions/runs/9423938980/job/25963266836)

If the e2e tests pass:

![Screenshot 2024-06-07 at 3 59 46 PM](https://github.com/woocommerce/woocommerce/assets/80276/0502a962-4a4e-4957-8064-725849e038e4)

The GH_TOKEN is getting set properly, and the step passes.  This is from [this test run](https://github.com/woocommerce/woocommerce/actions/runs/9423065580/job/25961499413)

I can't quite figure out why the environment variable would be set in one case but not in the other -- it looks like it should be set regardless.  Maybe a conditional elsewhere in `ci.yml`?

**Comment by adimoldovan** [comment]
>  It looks like it isn't getting the GH_TOKEN set properly

There is no token issue as far as I could see. The failure is caused by the wrong artifact name.

**Comment by adimoldovan** [comment]
I checked this and found 2 issues:

1. The allure-results folder is missing from the final uploaded archive so the workflow on woocommerce-test-reports side generates and empty report. I did not check to see when exactly this broke, but it's been like this for a while.
2. A more recent issue introduced with #48175, which hides the one above - the artifact name sent in the workflow dispatch is wrong and the final reports fail to download it and proceed with the report creation.
